### PR TITLE
feat: navbar functionality and styling - desktop version

### LINKED ISSUE #6: Top Navigation Styling
Style top navigation according to existing website. Consider mobile as well.

### PR DESCRIPTION
**Pull request by ekremney**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix https://github.com/hlxsites/elco-aerotop-sg/issues/6

Test URLs:
- Before: https://main--elco-aerotop-sg--hlxsites.hlx.page/de/
- After: https://issue-6-navbar--elco-aerotop-sg--hlxsites.hlx.page/de/

### CODE REVIEW

**Comment by solaris007** [comment]
i just saw the header box shadow on the original site, just FYI:
```
    box-shadow: 0 10px 50px -2px rgb(0 0 0 / 14%);
```

File: blocks/header/header.css [hunk 1]
```diff
@@ -170,10 +170,34 @@ header nav .nav-sections ul  {
...
+  border-top-style: solid;
+  transition-property: color,background-color,border-color;
+  transition-duration: .2s;
+  transition-timing-function: linear;
+}
+
+header nav .nav-sections ul li.active a,
+header nav .nav-sections ul li a:hover,
+header nav .nav-sections ul li.nav-drop:hover {
+  color: #bd3826;
```
Comment (solaris007) [inline blocks/header/header.css]: please use (or add) variable from base styles

File: blocks/header/header.css [hunk 2]
```diff
@@ -170,10 +170,34 @@ header nav .nav-sections ul  {
...
+  border-color: #bd3826;
```
Comment (solaris007) [inline blocks/header/header.css]: please use (or add) variable from base styles

File: blocks/header/header.css [hunk 3]
```diff
@@ -233,38 +239,46 @@ header nav .nav-sections ul > li > ul > li {
...
-    margin-top: 12px;
-    padding: 1em;
-    background-color: var(--highlight-background-color);
-    white-space: initial;
+    top: 100%;
+    right: 0;
+    width: 180px;
+    padding: 0;
+    background-color: #f2efef;
+    border-top: 3px solid #bd3826;
```
Comment (solaris007) [inline blocks/header/header.css]: please use (or add) variable from base styles

File: blocks/header/header.js [hunk 4]
```diff
@@ -112,17 +112,35 @@ export default async function decorate(block) {
...
+
     const navSections = nav.querySelector('.nav-sections');
     if (navSections) {
       navSections.querySelectorAll(':scope > ul > li').forEach((navSection) => {
-        if (navSection.querySelector('ul')) navSection.classList.add('nav-drop');
-        navSection.addEventListener('click', () => {
-          if (MQ.matches) {
-            const expanded = navSection.getAttribute('aria-expanded') === 'true';
+        const isDropdownList = navSection.querySelector('ul');
+        if (isDropdownList) navSection.classList.add('nav-drop');
```
Comment (solaris007) [inline blocks/header/header.js]: _nitpicking_, for readability please add a block in braces

File: blocks/header/header.js [hunk 5]
```diff
@@ -112,17 +112,35 @@ export default async function decorate(block) {
...
+
+        navSection.addEventListener('mouseleave', () => {
+          if (MQ.matches && isDropdownList) {
+            toggleAllNavSections(navSections);
+            navSection.setAttribute('aria-expanded', 'false');
+          }
+        });
+
+        // adds current css class to the current page in the navbar
+        const anchor = navSection.querySelector(':scope > a');
```
Comment (solaris007) [inline blocks/header/header.js]: i'd rename to `link` or similar, as anchor is usually referring to `#some-anchor` type links.

File: styles/styles.css [hunk 6]
```diff
@@ -10,6 +10,14 @@
  * governing permissions and limitations under the License.
  */
 
+@font-face {
```
Comment (solaris007) [inline styles/styles.css:13]: I added this in https://github.com/hlxsites/elco-aerotop-sg/pull/25

Comment (solaris007) [reply styles/styles.css:13]: maybe you didn't merge main right?

**Review by solaris007 — COMMENTED** [review]
great stuff, just small things

**Comment by cuzuco2** [comment]
I see some differences when the page goes to mobile sizes
original we have this:
image
And the branch shows this as the main content disappears:
image

File: styles/styles.css (hunk 6 above)
Comment (ekremney) [reply styles/styles.css:13]: I rebased by mistake. I think merging this PR won't create any problem

**Comment by ekremney** [comment]
> I see some differences when the page goes to mobile sizes original we have this: image And the branch shows this as the main content disappears: image

This PR contans the desktop navigation only. Mobile version will be included in the next PR

**Comment by solaris007** [comment]
vertical alignment of sub nav needs to be moved.

original:
Screenshot 2023-01-17 at 07 38 59

now:
Screenshot 2023-01-17 at 07 38 55

**Comment by solaris007** [comment]
there's a gap above the red hover/active marker, which is not present on the original:

Screenshot 2023-01-17 at 07 39 02

**Comment by solaris007** [comment]
there's a font difference.

original:
Screenshot 2023-01-17 at 07 42 18

now:
Screenshot 2023-01-17 at 07 42 11

**Comment by solaris007** [comment]
I don't see the box shadow applied.

original:
Screenshot 2023-01-17 at 07 44 51

**Review by solaris007 — APPROVED** [review]
love it

**Pull request merged**